homeassistant/components/metoffice/sensor.py

```diff
@@ -178,11 +180,52 @@ async def async_setup_entry(
     )
 
 
+def async_migrate_unique_ids(hass: HomeAssistant, hass_data: Any) -> None:
+    """Migrate unique ID for sensors based on name to key."""
+
+    for old_name, key in (
+        ("Station Name", "name"),
+        ("Weather", "weather"),
+        ("Temperature", "temperature"),
+        ("Feels Like Temperature", "feels_like_temperature"),
+        ("Wind Speed", "wind_speed"),
+        ("Wind Direction", "wind_direction"),
+        ("Wind Gust", "wind_gust"),
+        ("Visibility", "visibility"),
+        ("Visibility Distance", "visibility_distance"),
+        ("UV Index", "uv"),
+        ("Probability of Precipitation", "precipitation"),
+        ("Humidity", "humidity"),
+    ):
+        old_unique_id = f"{old_name}_{hass_data[METOFFICE_COORDINATES]}"
+        old_daily_unique_id = f"{old_unique_id}_{MODE_DAILY}"
+
+        ent_reg = entity_registry.async_get(hass)
+
+        if entity_id := ent_reg.async_get_entity_id(
+            Platform.SENSOR, DOMAIN, old_unique_id
+        ):
+            ent_reg.async_update_entity(
+                entity_id,
+                new_unique_id=f"{key}_{hass_data[METOFFICE_COORDINATES]}",
```

Comment:
I would remove the coordinates completely out of the unique ID and use the config entry ID instead.

Reply:
That would mean also migrating weather entities which is IMO unnecessary pain. Coordinates are an acceptable source of unique_id and they are also used as device identifier here.